source/particle/world.cc

```diff
@@ -1211,32 +1211,71 @@ namespace aspect
 
       const FiniteElement<dim> &velocity_fe = this->get_fe().base_element(this->introspection().base_elements.velocities);
 
+      const bool compute_fluid_velocity = this->include_melt_transport() && property_manager->get_data_info().fieldname_exists("melt_presence");
+      unsigned int fluid_component_index = numbers::invalid_unsigned_int;
+      std::vector<bool> use_fluid_velocity;
+      if (compute_fluid_velocity)
+        {
+          fluid_component_index = this->introspection().variable("fluid velocity").first_component_index;
+          const unsigned int melt_property_index = property_manager->get_data_info().get_position_by_field_name("melt_presence");
+
+          use_fluid_velocity.resize(particles_in_cell,false);
```

Comment:
You can also move this initialization into the declaration, the same way as for the `fluid_component_index`.